src/layer.js

```diff
@@ -216,4 +228,8 @@ Layer.prototype.draw = function(data) {
 			}
 		}
 	}
+
+	if (!selection.empty()) {
```

Comment:
This is for consistency with how we invoke the lifecycle events--see [this logic in `Layer#draw`](https://github.com/misoproject/d3.chart/blob/4e55cc67a77979d4a040ec18e3b9cd6303c5d6ad/src/layer.js#L186-L188).

...but it's not really fair to answer a question like, "Why are we doing this here?" with, "because we're doing it over there." The underlying motivation was originally discussed in this issue: https://github.com/misoproject/d3.chart/issues/69 .

Does this seem right to you?